pkg/services/dashboards/dashboard_service.go

```diff
@@ -15,13 +15,15 @@ import (
 type DashboardService interface {
 	SaveDashboard(dto *SaveDashboardDTO) (*models.Dashboard, error)
 	ImportDashboard(dto *SaveDashboardDTO) (*models.Dashboard, error)
+	DeleteDashboard(dashboardId int64, orgId int64) error
```

Comment:
I would rather put this in the `DashboardProvisioningService` interface instead to keep provisioning packages using only the `DashboardProvisioningService`. Reason we created the `DashboardProvisioningService` was that we had bugs mixing up validation/usage of functions of `DashboardService` between provisioning and regular dashboards.

It would probably make sense to have it in `DashboardService`, but then the places using DeleteDashboardCommand should use the `DashboardService` instead. Think we can wait with this for now.

Reply:
Hmm would that mean this would better be modelled as `DashboardService` being a subset of `DashboardProvisioningService` ? I mean in typescript I would say
```typescript
type DashboardProvisioningService = DashboardService & { ...functions specific to provisioning }
```

In Go would it make sense to declare the DashboardProvisioningService like this?
```go
type DashboardProvisioningService interface {
  DashboardService
  ...provisioning methods
}
```
then, if I get this correct, even the structs could be implemented as nested, having
```go
type dashboardProvisioningServiceImpl struct {
  *dashboardServiceImpl
  ...other stuff
}
```
To simplify the implementation, is that right?

Reply:
Or other possibility that would be maybe clearer is to have `DeleteProvisionedDashboard` on the `DashboardProvisioningService` that would at the moment just call `DeleteDashboard`? To have parity in the same way as `SaveDashboard/SaveProvisionedDashboard` right now.

Reply:
I went with using `dashboardProvisioningServiceImpl.DeleteProvisionedDashboard `, seems to me like that makes most sense.

Reply:
Awesome. I agree that makes sense.